sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidQuery.java

```diff
@@ -1687,10 +1688,27 @@ private Map<String, Object> withScanSignatureIfNeeded(
     if (!plannerContext.featureAvailable(EngineFeature.SCAN_NEEDS_SIGNATURE)) {
```

Comment:
Please also make a change in this patch so MSQ uses `scanQuery.getRowSignature()` if `CTX_SCAN_SIGNATURE` is not available. That provides us a path to get rid of `EngineFeature.SCAN_NEEDS_SIGNATURE` in the future.

Please also add a comment here about how it's kind of a legacy thing now, replaced by `columnTypes` in the scan query object.